extensions/spring-xd-extension-header-enricher/src/test/java/org/springframework/xd/header/enricher/HeaderEnricherIntegrationTest.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2015 the original author or authors.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.xd.header.enricher;
+
+import static org.junit.Assert.assertEquals;
+import static org.springframework.xd.dirt.test.process.SingleNodeProcessingChainSupport.chain;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import org.springframework.messaging.Message;
+import org.springframework.xd.dirt.server.singlenode.SingleNodeApplication;
+import org.springframework.xd.dirt.test.SingleNodeIntegrationTestSupport;
+import org.springframework.xd.dirt.test.process.SingleNodeProcessingChain;
+import org.springframework.xd.test.RandomConfigurationSupport;
+
+/**
+ * @author David Turanski
+ */
+public class HeaderEnricherIntegrationTest {
+
+	private static SingleNodeApplication application;
+
+	private static int RECEIVE_TIMEOUT = 5000;
+
+	private static String moduleName = "header-enricher";
+
+	SingleNodeProcessingChain chain;
+
+	/**
+	 * Start the single node container, binding random unused ports, etc. to not conflict with any other instances
+	 * running on this host. Configure the ModuleRegistry to include the project module.
+	 */
+	@BeforeClass
+	public static void setUp() {
+		System.setProperty("XD_HOME", "../..");
+		RandomConfigurationSupport randomConfigSupport = new RandomConfigurationSupport();
+		application = new SingleNodeApplication().run();
+		SingleNodeIntegrationTestSupport singleNodeIntegrationTestSupport = new SingleNodeIntegrationTestSupport(
+				application);
```

Comment:
These variables are not used (randomConfigSupport and singleNodeIntegrationTestSupport) and the tests run ok - is there some magic they provide that I am not aware of?